view/frontend/templates/button_product_page.phtml

```diff
@@ -0,0 +1,230 @@
+                                callback();
+                            }
+
+                            return false;
+                        }
+                        try {
+                            <?php echo $block->getjavascriptSuccess(); ?>
+                            callbacks.success_url = data.success_url;
+
+                            trackCallbacks.onSuccess();
+                        } finally {
+                            callback();
+                        }
+                    };
+                    // ajax call to the update order transaction data endpoint.
+                    // passes the bolt transaction reference
+                    save_request = ajaxPost(settings.save_order_url, parameters, onSuccess);
+                },
+
+                check: function () {
+                    /**
+                     * On Bolt button click check if guest checkout is allowed.
+                     * Display login popup to guest customers if it is not. The
+                     * Magento customerData and authenticationPopup objects are
+                     * used.
+                     */
+
+                    //check if we have inventory on stock
+                    if ($qty && (Number($qty.val()) > max_qty)) {
+                        window.alert("Please correct the quantity. The most you may purchase is " + max_qty + ".");
+                        return false;
+                    }
+
+                    // check if login is required
+                    if ( !customer().firstname && !shoppingCart().isGuestCheckoutAllowed) {
+                        // if authentication is required for checkout set a cookie
+                        // for auto opening Bolt checkout after login
+                        authenticationPopup.showModal();
+                        return false;
+                    }
+                    return true;
+                },
+
+                onEmailEnter: function(email) {
+                    trackCallbacks.onEmailEnter(email);
+                    if (callbacks.email !== email) {
+                        ajaxPost(settings.save_email_url, 'email='+encodeURIComponent(email));
+                        callbacks.email = email;
+                    }
+                }
+            };
+
+            function setupProductPage() {
+                var quantity = 1;
+                if ($qty) {
+                    quantity = Number($qty.val());
+                }
+                var cart = {
+                    currency: "USD",
+                    items: [{
+                        reference: '<?= $block->getProduct()->getId() ?>',
+                        price: <?= (int) $block->getProduct()->getPrice() ?>,
+                        name: '<?= $block->getProduct()->getName() ?>',
+                        quantity: quantity,
+                    }]
+                };
+
+                var buttonClass = "bolt-product-checkout-button";
+                var hints = {prefill:{}};
```

Comment:
maybe add TODO fo adding prefill based on logged in user info?

Reply:
I created a separate task for it